x-pack/plugins/observability_solution/observability_ai_assistant/server/rule_connector/index.ts

```diff
@@ -249,28 +261,43 @@ export const getObsAIAssistantConnectorAdapter = (): ConnectorAdapter<
         connector: params.connector,
         message: params.message,
         rule: { id: rule.id, name: rule.name, tags: rule.tags, ruleUrl: ruleUrl ?? null },
-        alertState:
-          alerts.new.count > 0
-            ? 'new'
-            : alerts.ongoing.count > 0
-            ? 'ongoing'
-            : alerts.recovered.count > 0
-            ? 'recovered'
-            : 'unknown',
+        alertSummary: getAlertSummary(alerts),
       };
     },
   };
 };
 
-function getBackgroundProcessInstruction(rule: RuleType, state: AlertState) {
-  let instruction = `You are called as a background process because the alert ${JSON.stringify(
+function getBackgroundProcessInstruction(rule: RuleType, summary: AlertSummary) {
+  let instruction = `You are called as a background process because the following alerts have changed state for the rule ${JSON.stringify(
     rule
-  )} ${state === 'new' ? 'fired' : 'recovered'}.`;
+  )}:\n`;
+  if (summary.new.length > 0) {
+    instruction += `- ${summary.new.length} alerts have fired: ${JSON.stringify(summary.new)}\n`;
+  }
+  if (summary.recovered.length > 0) {
+    instruction += `- ${summary.recovered.length} alerts have recovered: ${JSON.stringify(
+      summary.recovered
+    )}\n`;
+  }
   instruction +=
     ' As a background process you are not interacting with a user. Because of that DO NOT ask for user';
   instruction +=
     ' input if tasked to execute actions. You can generate multiple responses in a row.';
-  instruction += ' If available, include the link to the conversation at the end of your answer.';
+  instruction += ' If available, include the link of the conversation at the end of your answer.';
 
   return instruction;
 }
+
+function getAlertSummary(alerts: CombinedSummarizedAlerts): AlertSummary {
+  const extractAlertData = (hit: AlertHit) => {
+    return {
+      id: hit._id,
+      reason: get(hit, 'kibana.alert.reason') || null,
```

Comment:
any reason why we would not send over the full alert data here? Maybe good to talk to @sorenlouv about this as well - we can also do it in a followup.

Reply:
I wanted to limit the noise and amount of token sent but that's probably too early for that. Updated in https://github.com/elastic/kibana/pull/179980/commits/cfe88002f338b7c2287ea73f88e3f3f88994d805